mlserver/model.py

```diff
@@ -74,11 +74,24 @@ async def predict(self, payload: InferenceRequest) -> InferenceResponse:
         Method responsible for running inference on the model.
 
 
-        **This method should be overriden to implement your custom inference
+        **This method can be overriden to implement your custom inference
         logic.**
         """
         raise NotImplementedError("predict() method not implemented")
 
+    async def predict_stream(
+        self, payloads: AsyncIterator[InferenceRequest]
+    ) -> AsyncIterator[InferenceResponse]:
+        """
+        Method responsible for running generation on the model, streaming a set
+        of responses back to the client.
+
+
+        **This method can be overriden to implement your custom inference
+        logic.**
+        """
+        yield await self.predict((await payloads.__anext__()))
```

Comment:
if the user uses `predict_stream` without implementing it, we will do infer on the first element?

should we raise `NotImplementedError` similar to the pattern in `predict`?

Reply:
I think this default behaviour should be fine. It was also implemented in the first attempt to introduce streaming support.